src/Core/Content/Product/Subscriber/ProductSubscriber.php

```diff
@@ -50,7 +50,7 @@ private function sortProperties(SalesChannelProductEntity $product): PropertyGro
         foreach ($properties as $option) {
             $origin = $option->getGroup();
 
-            if (!$origin) {
+            if (!$origin || !$origin->getVisibleOnProductDetailPage()) {
```

Comment:
Can you try to limit the SQL result directly? Here: https://github.com/shopware/platform/blob/963b38bbd87015f0036494b3d4065cf5541553b4/src/Storefront/Page/Product/ProductLoader.php#L40

Should be something like

$criteria->getAssociation('options')->addFilter(new EqualsFilter('visibleOnProductDetailPage', true));

Reply:
No, a developer might want/need the option in the 'options' property of the product. I think it should only be filtered in the sortedOptions.